catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -466,4 +544,26 @@ private boolean isStorageAvailable(StorageProvider storageProvider) {
         }
         return true;
     }
+
+    private void logFailedQueryInfo(DeleteRequest deleteRequest, String attributeName,
+            QueryResponse query) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(
+                    "While rewriting the query, did not get a metacardId corresponding to every attribute."
+                            + "Original Delete By attribute was: " + attributeName);
+            LOGGER.debug("Metacards unable to get Metacard ID from are: "
+                    + deleteRequest.getAttributeValues()
+                    .stream()
+                    .map(Object::toString)
+                    .filter(s -> !query.getResults()
```

Comment:
You're recreating this set for each attributeValue. Extract the logic to create the set and change this predicated to 
`.filter(s -> !efficientlyCreatedSet.contains(s))`